airtable/emails.js

```diff
@@ -0,0 +1,229 @@
+                allMessages.find(m => m.id === existingMessage.id).getCellValue("Email type").name
+            );
+        } else {
+            existingMessageTypes = [];
+        }
+
+        // create a new message for the volunteer if necessary
+        if (existingMessageTypes.indexOf("Matched: to Volunteer") === -1) {
+            const volunteerEmail = sendgridData.volunteer.email;
+            if (volunteerEmail) {
+                messagesToCreate.push({
+                    fields: {
+                        "Delivery": [{ "id": delivery.id }],
+                        "Email type": { "name": "Matched: to Volunteer" },
+                        "Status": { name: "Queued" },
+                        "Recipient": volunteerEmail,
+                        "Template Data": JSON.stringify(sendgridData)
+                    }
+                })
+            } else {
+                output.markdown(`**Warning, Missing email address**: couldn't create message for Volunteer ${volunteer.getCellValue("First Name")} ${volunteer.getCellValue("Last Name")}`)
+            }
+        }
+
+        // create a new message for the recipient if necessary
+        if (existingMessageTypes.indexOf("Matched: to Recipient") === -1) {
+            const recipientEmail = sendgridData.delivery.email;
+            if (recipientEmail) {
+                messagesToCreate.push({
+                    fields: {
+                        "Delivery": [{ "id": delivery.id }],
+                        "Email type": { "name": "Matched: to Recipient" },
+                        "Status": { name: "Queued" },
+                        "Recipient": recipientEmail,
+                        "Template Data": JSON.stringify(sendgridData)
+                    }
+                })
+            } else {
+                output.markdown(`**Warning, Missing email address**: Couldn't create message for delivery recipient ${delivery.name}.`)
+            }
+        }
+    }
+
+    return messagesToCreate;
+}
+
+// Given a Message Record from our Airtable, send it using Sendgrid
+async function sendMessage(messageToSend) {
+    const dynamicTemplateData = JSON.parse(messageToSend.getCellValue("Template Data"));
+    const sendgridData = {
+        "personalizations": [
+            {
+                "to": [
+                    {
+                    "email": messageToSend.getCellValue("Recipient")
+                    }
+                ],
+                "dynamic_template_data": dynamicTemplateData
+            }
+        ],
+        "from": {
+            "name": "Neighbor Express",
+            "email": "noreply@neighborexpress.org"
+        },
+        "reply_to": REPLY_TO,
+        "template_id": SENDGRID_TEMPLATES[messageToSend.getCellValue("Email type").name]
+    }
+
+    const response = await fetch("https://nex-sendgrid-proxy.geoffreylitt.now.sh/api/sendgrid-proxy?proxy_token=nqotguhnBkZph52GJ55JHtWRlw4H2xHGZqfHLHusLpt", {
```

Comment:
todo: move the nex-sendgrid-proxy into the nex zeit org (had trouble with this first time around for some reason)